platform/s3_bucket/main.tf

```diff
@@ -84,6 +84,13 @@ resource "aws_s3_object" "reqs" {
   etag   = filemd5(var.local_requirement_file_path)
 }
 
+resource "aws_s3_object" "startup_script" {
```

Comment:
@amarouane-ABDELHAK I haven't worked with TF in a while. How would I make this optional, if the user doesn't provide a startup script? Don't want to introduce a backward incompatibility unnecessarily.

Reply:
we can use `count` parameter
```
resource "aws_s3_object" "startup_script" {
 count =  var.local_startup_script_file_path == null ? 0: 1
  bucket = aws_s3_bucket.this.id
  key    = "requirements/${var.startup_script_filename}"
  source = var.local_startup_script_file_path
  etag   = filemd5(var.local_startup_script_file_path)
}
```
assuming `var.local_startup_script_file_path` is optional and its default is null

Reply:
Ok, thanks. Will update.